src/lib.rs

```diff
@@ -71,12 +75,21 @@ mod wrap_main {
     }
 }
 
-pub fn run_main(context: &Context, args: impl Iterator<Item = String>) -> R<ExitCode> {
-    Ok(match cli::parse_args(args)? {
+pub fn run_main(context: &Context, args: &cli::Args) -> R<ExitCode> {
+    Ok(match args {
         cli::Args::ExecutableMock {
             executable_mock_path,
         } => executable_mock::run(context, &executable_mock_path)?,
-        cli::Args::CheckProtocols { script_path } => run_check_protocols(context, &script_path)?,
+        cli::Args::CheckProtocols {
```

Comment:
not sure if it's an improvement, but reminder that you can do
```rust
        cli::Args::CheckProtocols {
            script_path,
            record: true,
        } => print_recorded_protocol(context, script_path)?,
        cli::Args::CheckProtocols { script_path, .. } => {
            run_check_protocols(context, &script_path)?
        }
```

Reply:
Good idea. I tried that and I actually think it looks slightly worse, though.